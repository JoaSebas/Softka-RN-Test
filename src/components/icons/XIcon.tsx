import React from 'react';
import Svg, { Line } from 'react-native-svg';
import { colors } from '../../styles/colors';

const XIcon = ({ size = 24, color = colors.textLight }) => {
  return (
    <Svg  testID="x-icon" width={size} height={size} viewBox="0 0 24 24" fill="none" stroke={color} strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
      <Line x1="4" y1="4" x2="15" y2="15" />
      <Line x1="15" y1="4" x2="4" y2="15" />
    </Svg>
  );
};

export default XIcon;
