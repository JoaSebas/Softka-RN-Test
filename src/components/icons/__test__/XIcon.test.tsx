import React from 'react';
import { render } from '@testing-library/react-native';
import XIcon from '../XIcon';

describe('XIcon', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(<XIcon />);
    expect(getByTestId('x-icon')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = render(<XIcon size={32} color="red" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
