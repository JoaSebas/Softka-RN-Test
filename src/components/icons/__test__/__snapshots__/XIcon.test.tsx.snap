// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XIcon should match snapshot 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={32}
  bbWidth={32}
  fill="none"
  focusable={false}
  height={32}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="red"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 32,
        "width": 32,
      },
    ]
  }
  testID="x-icon"
  vbHeight={24}
  vbWidth={24}
  width={32}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
        "strokeLinecap",
        "strokeLinejoin",
      ]
    }
    stroke={
      {
        "payload": 4294901760,
        "type": 0,
      }
    }
    strokeLinecap={1}
    strokeLinejoin={1}
    strokeWidth="2"
  >
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="4"
      x2="15"
      y1="4"
      y2="15"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="15"
      x2="4"
      y1="4"
      y2="15"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
